internal/serialization/object_data.go

```diff
@@ -312,779 +317,307 @@ func (i *ObjectDataInput) SetPosition(pos int32) {
 }
 
 func (i *ObjectDataInput) ReadByte() byte {
-	if i.err != nil {
-		return 0
-	}
-	var ret byte
-	ret, i.err = i.readByte()
-	return ret
+	i.AssertAvailable(ByteSizeInBytes)
+	return i.readByte()
 }
 
-func (i *ObjectDataInput) readByte() (byte, error) {
-	var err = i.AssertAvailable(bufutil.ByteSizeInBytes)
-	var ret byte
-	if err == nil {
-		ret = ReadUInt8(i.buffer, i.position)
-		i.position += bufutil.ByteSizeInBytes
-	}
-	return ret, err
-}
-
-func (i *ObjectDataInput) ReadByteWithPosition(pos int32) byte {
-	if i.err != nil {
-		return 0
-	}
-	var res byte
-	res, i.err = i.readByteWithPosition(pos)
-	return res
-}
-
-func (i *ObjectDataInput) readByteWithPosition(pos int32) (byte, error) {
-	var err = i.AssertAvailable(bufutil.ByteSizeInBytes)
-	var ret byte
-	if err == nil {
-		ret = ReadUInt8(i.buffer, pos)
-	}
-	return ret, err
-}
-
-func (i *ObjectDataInput) ReadBool() bool {
-	if i.err != nil {
-		return false
-	}
-	var ret bool
-	ret, i.err = i.readBool()
+func (i *ObjectDataInput) readByte() byte {
+	ret := i.buffer[i.position]
+	i.position += ByteSizeInBytes
 	return ret
 }
 
-func (i *ObjectDataInput) readBool() (bool, error) {
-	var err = i.AssertAvailable(bufutil.BoolSizeInBytes)
-	var ret bool
-	if err == nil {
-		ret = ReadBool(i.buffer, i.position)
-		i.position += bufutil.BoolSizeInBytes
-	}
-	return ret, err
+func (i *ObjectDataInput) ReadByteAtPosition(pos int32) byte {
+	return i.buffer[pos]
```

Comment:
nit: we may want to include a range check into this and similar position based methods

Reply:
All we can do is panic in case we find out pos is out of range. A range check there would be probably quite costly, but I'll check what we can do about that.